pkg/ipam/crd.go

```diff
@@ -175,25 +175,37 @@ func newNodeStore(nodeName string, owner Owner) *nodeStore {
 		0,
 		cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
+				var valid, equal bool
+				defer func() { owner.K8sEventReceived("CiliumNode", "create", valid, equal) }()
 				if node, ok := obj.(*ciliumv2.CiliumNode); ok {
+					valid = true
 					log.Debugf("New CiliumNode %+v", node)
 					store.updateLocalNodeResource(node)
+					owner.K8sEventProcessed("CiliumNode", "create", true)
```

Comment:
Now I understand why the daemon is the receiver, maybe this `crd.go` should go under the `pkg/k8s` as well as the metrics function.

Reply:
Covered in comment below